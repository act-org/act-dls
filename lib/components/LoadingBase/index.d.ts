/**
 * @prettier
 */
import * as React from 'react';
import { Props as CircularProgressBaseProps } from '~/components/CircularProgressBase';
export interface Props {
    circularProgressProps?: CircularProgressBaseProps;
    style?: React.CSSProperties;
    title?: string;
}
declare const LoadingBase: React.FC<Props>;
export default LoadingBase;
//# sourceMappingURL=index.d.ts.map