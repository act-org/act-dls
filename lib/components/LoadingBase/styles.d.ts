/**
 * @prettier
 */
declare const _default: (props?: any) => Record<"typographyRoot", string>;
export default _default;
//# sourceMappingURL=styles.d.ts.map