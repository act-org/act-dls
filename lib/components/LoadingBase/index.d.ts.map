{"version":3,"file":"index.d.ts","sourceRoot":"","sources":["../../../src/components/LoadingBase/index.tsx"],"names":[],"mappings":"AAAA;;GAEG;AAEH,OAAO,KAAK,KAAK,MAAM,OAAO,CAAC;AAG/B,OAA6B,EAC3B,KAAK,IAAI,yBAAyB,EACnC,MAAM,mCAAmC,CAAC;AAG3C,MAAM,WAAW,KAAK;IACpB,qBAAqB,CAAC,EAAE,yBAAyB,CAAC;IAClD,KAAK,CAAC,EAAE,KAAK,CAAC,aAAa,CAAC;IAC5B,KAAK,CAAC,EAAE,MAAM,CAAC;CAChB;AAED,QAAA,MAAM,WAAW,EAAE,KAAK,CAAC,EAAE,CAAC,KAAK,CAwChC,CAAC;AAEF,eAAe,WAAW,CAAC"}