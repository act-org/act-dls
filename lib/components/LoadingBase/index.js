"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _style = _interopRequireDefault(require("styled-jsx/style"));

var React = _interopRequireWildcard(require("react"));

var _CircularProgressBase = _interopRequireDefault(require("../CircularProgressBase"));

var _TypographyBase = _interopRequireDefault(require("../TypographyBase"));

var _styles = _interopRequireDefault(require("./styles"));

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function _getRequireWildcardCache() { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

var LoadingBase = function LoadingBase(_ref) {
  var circularProgressProps = _ref.circularProgressProps,
      style = _ref.style,
      title = _ref.title;
  var classes = (0, _styles["default"])();
  return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(_style["default"], {
    id: "3586607810"
  }, [".container.jsx-3586607810{-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;height:100%;-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;width:100%;}"]), /*#__PURE__*/React.createElement("div", {
    "data-testid": "loading",
    style: style,
    className: "jsx-3586607810" + " " + "container"
  }, /*#__PURE__*/React.createElement(_CircularProgressBase["default"], _extends({
    size: 24,
    thickness: 4.5
  }, circularProgressProps)), title && /*#__PURE__*/React.createElement(_TypographyBase["default"], {
    classes: {
      root: classes.typographyRoot
    },
    variant: "overline"
  }, title)));
};

var _default = LoadingBase;
exports["default"] = _default;