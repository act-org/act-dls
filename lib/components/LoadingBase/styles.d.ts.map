{"version":3,"file":"styles.d.ts","sourceRoot":"","sources":["../../../src/components/LoadingBase/styles.ts"],"names":[],"mappings":"AAAA;;GAEG;;AAMH,wBAIG"}