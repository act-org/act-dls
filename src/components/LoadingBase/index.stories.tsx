/**
 * @prettier
 */

import * as React from 'react';
import { text } from '@storybook/addon-knobs';

import LoadingBase from '.';

export const Base = (): React.ReactElement<any> => (
  <LoadingBase
    circularProgressProps={{
      size: Number(text('Size', '24')),
      thickness: Number(text('Thickness', '4.5')),
    }}
    title={text('Title', 'Loading...')}
  />
);

export default {
  component: LoadingBase,
  parameters: {
    info: {
      text: 'This is the base variant of the Loading component.',
    },
  },
  title: 'Molecules/Loading',
};
