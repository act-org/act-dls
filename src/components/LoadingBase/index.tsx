/**
 * @prettier
 */

import * as React from 'react';
import { Typography } from '@material-ui/core';

import CircularProgressBase, {
  Props as CircularProgressBaseProps,
} from '~/components/CircularProgressBase';
import useStyles from './styles';

export interface Props {
  circularProgressProps?: CircularProgressBaseProps;
  style?: React.CSSProperties;
  title?: string;
}

const LoadingBase: React.FC<Props> = ({
  circularProgressProps,
  style,
  title,
}: Props): React.ReactElement<any> => {
  const classes = useStyles();

  return (
    <>
      <style jsx>
        {`
          .container {
            align-items: center;
            display: flex;
            flex-direction: column;
            height: 100%;
            justify-content: center;
            width: 100%;
          }
        `}
      </style>

      <div className="container" data-testid="loading" style={style}>
        <CircularProgressBase
          size={24}
          thickness={4.5}
          {...circularProgressProps}
        />

        {title && (
          <Typography
            classes={{ root: classes.typographyRoot }}
            variant="overline"
          >
            {title}
          </Typography>
        )}
      </div>
    </>
  );
};

export default LoadingBase;
