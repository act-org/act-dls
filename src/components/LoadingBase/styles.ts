/**
 * @prettier
 */

import { makeStyles } from '@material-ui/core/styles';

import DIMS from '~/constants/dims';

export default makeStyles({
  typographyRoot: {
    marginTop: DIMS.LAYOUT_PADDING / 2,
  },
});
