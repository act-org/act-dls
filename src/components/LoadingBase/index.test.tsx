/**
 * @prettier
 */

import * as React from 'react';

import render from '~/helpers/test/render';

import LoadingBase from '.';

describe('LoadingBase', () => {
  it('matches the snapshot', () => {
    const { container } = render(<LoadingBase title="Loading..." />);

    expect(container).toMatchSnapshot();
  });
});
