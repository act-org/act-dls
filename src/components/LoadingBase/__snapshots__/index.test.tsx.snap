// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingBase matches the snapshot 1`] = `
<div>
  <style>
    
          .container {
            align-items: center;
            display: flex;
            flex-direction: column;
            height: 100%;
            justify-content: center;
            width: 100%;
          }
        
  </style>
  <div
    class="container"
    data-testid="loading"
  >
    <div
      class="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
      role="progressbar"
      style="width: 24px; height: 24px;"
    >
      <svg
        class="MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
          cx="44"
          cy="44"
          fill="none"
          r="19.75"
          stroke-width="4.5"
        />
      </svg>
    </div>
    <span
      class="MuiTypography-root makeStyles-typographyRoot-1 MuiTypography-overline"
    >
      Loading...
    </span>
  </div>
</div>
`;
